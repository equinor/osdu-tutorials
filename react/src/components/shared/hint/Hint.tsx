import React from 'react';
import './styles.css';

/**
 * An interface of the props Hint React component
 * @param {string} title - hint main text
 * @param {string} subTitle - hint additional text
 */
export interface HintProps {
  /** upper text, bigger font size  */
  title?: string;

  /** lower text, smaller font size */
  subTitle?: string;

  /** a small size for a more compact view  */
  size?: 'normal' | 'small';
}

/**
 * Create simple hint
 * @param {HintProps} props
 */
export function Hint({ title, subTitle, size = 'normal' }: HintProps) {
  return (
    <div className={`hint hint--size_${size}`}>
      <div className="hint__text hint__title">{title}</div>
      <div className="hint__text hint__sub-title">{subTitle}</div>
    </div>
  );
}
