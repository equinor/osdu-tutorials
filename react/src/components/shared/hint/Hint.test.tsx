import React from 'react';
import { shallow } from 'enzyme';
import { Hint, HintProps } from './Hint';

describe('Hint', () => {
  it('should render self and subcomponents', () => {
    const props: HintProps = {
      title: 'Test title',
      subTitle: 'Test subtitle',
    };
    const wrapper = shallow(<Hint {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
