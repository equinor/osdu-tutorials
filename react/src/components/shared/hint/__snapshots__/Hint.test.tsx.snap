// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hint should render self and subcomponents 1`] = `
<div
  className="hint hint--size_normal"
>
  <div
    className="hint__text hint__title"
  >
    Test title
  </div>
  <div
    className="hint__text hint__sub-title"
  >
    Test subtitle
  </div>
</div>
`;
